### PR TITLE
[Identity] Propagate send_certificate_chain in OBO

### PR DESCRIPTION
**Pull request by pvaneck**
Closes: https://github.com/Azure/azure-sdk-for-python/issues/36489

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_credentials/on_behalf_of.py [hunk 1]
```diff
@@ -98,7 +101,10 @@ def __init__(
                 raise ValueError('Specifying both "client_certificate" and "client_secret" is not valid.')
             try:
                 credential = get_client_credential(
-                    certificate_path=None, password=kwargs.pop("password", None), certificate_data=client_certificate
+                    certificate_path=None,
+                    password=kwargs.pop("password", None),
+                    certificate_data=client_certificate,
+                    send_certificate_chain=kwargs.pop("send_certificate_chain", False),
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_credentials/on_behalf_of.py:107]: Why not put it into the signature?

File: sdk/identity/azure-identity/azure/identity/_credentials/on_behalf_of.py [hunk 2]
```diff
@@ -98,7 +101,10 @@ def __init__(
                 raise ValueError('Specifying both "client_certificate" and "client_secret" is not valid.')
             try:
                 credential = get_client_credential(
-                    certificate_path=None, password=kwargs.pop("password", None), certificate_data=client_certificate
+                    certificate_path=None,
+                    password=kwargs.pop("password", None),
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_credentials/on_behalf_of.py:105]: Same here.

File: sdk/identity/azure-identity/azure/identity/_credentials/on_behalf_of.py (hunk 1 above)
Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/_credentials/on_behalf_of.py]: Updated.

**Review by xiangyan99 — APPROVED** [review]
Please also check the async one.

**Pull request merged**